### DISCUSSION TITLE
@nx:js executors don't allow copying assets from gitignored folders

### DISCUSSION THREAD

**Discussion by b-smets** [nrwl/nx#23293] [Feature Requests]
### Current Behavior

The @nx:js executors have an option to copy assets over to the build artifacts of the current project being built. However, this automatically ignores any files that match a pattern in the .gitignore / .nxignore files. Because of this, it is impossible to have a lib contribute assets to another lib / app if this assset is a build artifact.

In our setup we have a lib that produces a CSS file based on some SCSS input files. The CSS file is a build artifact.
We would like to consume this CSS file as an asset in another lib, using the assets of the @nx:js executor.

The culprits are the following lines:

```
    this.ignore = ignore();
    const gitignore = path.join(opts.rootDir, '.gitignore');
    const nxignore = path.join(opts.rootDir, '.nxignore');
    if (fse.existsSync(gitignore))
      this.ignore.add(fse.readFileSync(gitignore).toString());
    if (fse.existsSync(nxignore))
      this.ignore.add(fse.readFileSync(nxignore).toString());
```
https://github.com/nrwl/nx/blob/master/packages/js/src/utils/assets/copy-assets-handler.ts#L68

```
      if (
        !assetGlob.ignore?.some((ig) => minimatch(src, ig)) &&
        !this.ignore.ignores(src)
      ) {
```
https://github.com/nrwl/nx/blob/master/packages/js/src/utils/assets/copy-assets-handler.ts#L185

### Expected Behavior

I would expect this behaviour to be configurable. For example, an option could be added to the asset object to instruct NX to not ignore .gitignored files.

### Steps to Reproduce

1. Create a lib that produces an asset as a build artifact
2. Create a second lib and reference the build artifact of the first lib in the assets section of the @nx/js build executor.

### Nx Report

```shell
nx report

 >  NX   Report complete - copy this into the issue template

   Node   : 18.19.0
   OS     : darwin-x64
   yarn   : 1.22.19
   
   nx (global)        : 18.0.3
   nx                 : 18.0.3
   @nx/js             : 18.0.3
   @nx/jest           : 18.0.3
   @nx/linter         : 18.0.3
   @nx/eslint         : 18.0.3
   @nx/workspace      : 18.0.3
   @nx/cypress        : 18.0.3
   @nx/devkit         : 18.0.3
   @nx/esbuild        : 18.0.3
   @nx/eslint-plugin  : 18.0.3
   @nx/node           : 18.0.3
   @nx/plugin         : 18.0.3
   @nx/react          : 18.0.3
   @nrwl/tao          : 18.0.3
   @nx/vite           : 18.0.3
   @nx/web            : 18.0.3
   @nx/webpack        : 18.0.3
   typescript         : 5.3.3
   ---------------------------------------
   Local workspace plugins:
         @shared/executors
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by mxvsh** [comment]
no fix on this?

**Reply by raffidil** [reply]
This temporary solution worked for me:
https://github.com/nrwl/nx/issues/6821#issuecomment-1322761185

**Comment by alexandr2110pro** [comment]
Hey there. Really annoying. Imagine you have monorepo with multiple prisma libs (each with its own schema and client).
You will end up generating the client into the workspace library which, of course, you want to gitignore.

Now, when you build an app that uses one of these libs, you want to copy the `.node` and `.prisma` files. But the dir that contains them is ignored!

For now, we have to install the prisma and generate the client inside of the container (in Dockerfile), which adds extra 30-40 seconds to every build in the CI.

**Comment by kuus** [comment]
hi, same problem here with multiple apps sharing the same _static assets_ to be copied over each app's `public` folder

**Comment by kenjiqq** [comment]
Also same problem if you want to copy assets from a node_modules installed package, since you will have node_modules in your gitignore file
